test/sample_test.yaml

```diff
@@ -19,184 +19,120 @@ metadata:
-            value: "{{inputs.parameters.target-image-prefix}}{{inputs.parameters.sample-tests-image-suffix}}"
-          - name: namespace
-            value: "{{inputs.parameters.namespace}}"
-          - name: test-name
-            value: "xgboost"
-      - name: run-notebook-tfx-tests
-        template: run-sample-tests
-        arguments:
-          parameters:
-          - name: test-results-gcs-dir
-            value: "{{inputs.parameters.test-results-gcs-dir}}"
+        value: kubeflow
+  templates:
+    - name: sample-test
+      inputs:
+        parameters:
           - name: target-image-prefix
-            value: "{{inputs.parameters.target-image-prefix}}"
-          - name: sample-tests-image
-            value: "{{inputs.parameters.target-image-prefix}}{{inputs.parameters.sample-tests-image-suffix}}"
-          - name: namespace
-            value: "{{inputs.parameters.namespace}}"
-          - name: test-name
-            value: "notebook-tfx"
-      - name: run-notebook-lightweight-tests
-        template: run-sample-tests
-        arguments:
-          parameters:
           - name: test-results-gcs-dir
-            value: "{{inputs.parameters.test-results-gcs-dir}}"
-          - name: target-image-prefix
-            value: "{{inputs.parameters.target-image-prefix}}"
-          - name: sample-tests-image
-            value: "{{inputs.parameters.target-image-prefix}}{{inputs.parameters.sample-tests-image-suffix}}"
+          - name: sample-tests-image-suffix
           - name: namespace
-            value: "{{inputs.parameters.namespace}}"
-          - name: test-name
-            value: "notebook-lightweight"
-      - name: run-notebook-typecheck-tests
-        template: run-sample-tests
-        arguments:
-          parameters:
+      steps:
+        - - name: build-sample-tests-image
+            template: build-image-by-dockerfile
+            arguments:
+              parameters:
+                - name: docker-path
+                  value: .
+                - name: docker-file
+                  value: test/sample-test/Dockerfile
+                - name: image-name
+                  value: "{{inputs.parameters.target-image-prefix}}{{inputs.parameters.sample-tests-image-suffix}}"
+        - - name: run-sample-tests-loop
+            template: run-sample-tests
+            arguments:
+              parameters:
+                - name: namespace
+                  value: "{{inputs.parameters.namespace}}"
+                - name: sample-tests-image
+                  value: "{{inputs.parameters.target-image-prefix}}{{inputs.parameters.sample-tests-image-suffix}}"
+                - name: target-image-prefix
+                  value: "{{inputs.parameters.target-image-prefix}}"
+                - name: test-results-gcs-dir
+                  value: "{{inputs.parameters.test-results-gcs-dir}}"
+                - name: test-name
+                  value: "{{item.test-name}}"
+            withItems:
```

Comment:
Is there particular reason to use a list of dict? 
How about the following
withItems:
  - "tf-training"
  - "tfx"
If we need multiple params besides the test name in the future, we can convert it to a dict.
WDYT?

Reply:
Totally agree with you that currently we can do a list of str given the simple structure of params we have.
I chose dict cause I personally found the name 'item' is not very meaningful. PTAL.